app/Services/OfferService.php

```
@@ -0,0 +1,120 @@
+                //TODO: main image can be deleted 
+            }
+        }
+    }
+
+    public function getPaginatedOffers(int $lenght, array $filters = [], ?Collection $dynamicFilters = null): LengthAwarePaginator
+    {
+        $user = Auth::user() ?? null;
+        return Offer::with('brand')
+            ->filter($filters)
+            ->when(!empty($dynamicFilters), function ($query) use ($dynamicFilters) {
+                foreach ($dynamicFilters as $key => $value) {
```

PHPMD (2.15.0)

Avoid unused local variables such as '$key'. (Unused Code Rules)

(UnusedLocalVariable)